### ISSUE TITLE
MGOS_SHADOW_UPDATE_REJECTED event not raised

### ISSUE THREAD

**Issue by ghost** [mongoose-os-libs/aws#6]
The _AWS Shadow Service_ rejects an update because it contains invalid json, but the `MGOS_SHADOW_UPDATE_REJECTED` event is not raised (see log below).

```
[Jul 25 09:37:18.446] mgos_mqtt_global_con MQTT connecting to a2i6vvtwfifj27.iot.eu-west-2.amazonaws.com:8883
[Jul 25 09:37:18.669] mongoose_poll        New heap free LWM: 35240
[Jul 25 09:37:18.773] mongoose_poll        New heap free LWM: 34712
[Jul 25 09:37:24.457] mongoose_poll        New heap free LWM: 15400
[Jul 25 09:37:24.605] mgos_mqtt_ev         MQTT TCP connect ok (0)
[Jul 25 09:37:24.700] mgos_mqtt_ev         MQTT CONNACK 0
[Jul 25 09:37:24.780] mgos_aws_shadow_ev   Subscribed
[Jul 25 09:37:24.785] zt_shadow_handle_ev  Shadow Event: 1396987650 (CONNECTED)
[Jul 25 09:37:24.791] zt_shadow_handle_ev  Shadow Event: 1396987649 (UPDATE)
[Jul 25 09:37:24.797] mgos_aws_shadow_ev   Requesting state, current version 0
[Jul 25 09:37:24.804] mgos_aws_shadow_ev   Update: {"state": {"reported": { "dht": { "t": nan, "h": nan } }}, "clientToken": "b0aedebc"}
[Jul 25 09:37:24.908] mgos_aws_shadow_ev   Version: 0 -> 3 (2)
[Jul 25 09:37:24.913] zt_shadow_handle_ev  Shadow Event: 1396987651 (GET_ACCEPTED)
[Jul 25 09:37:24.922] zt_shadow_handle_ev  Shadow Event: 1396987656 ()
[Jul 25 09:37:24.926] mgos_aws_shadow_ev   Error: 400 Payload contains invalid json
[Jul 25 09:37:25.737] pm open,type:2 0
```
As you can see, the shadow's event handler (`zt_shadow_handle_ev`), receives following events:

1. CONNECTED
2. UPDATE
3. GET_ACCEPTED (because the automatic GET on (re)connect)
4. ??? (event 1396987656)

**The handler doesn't receive the UPDATE_REJECTED event, even if the AWS raised it** (see picture below).
![image](https://user-images.githubusercontent.com/25845078/43189473-f297628e-8ff6-11e8-8331-06a27e848fba.png)

Please refer to the source code tested on an **ESP8266 WeMos D1 mini**.
[dht-aws-lowpower-sensor-c.zip](https://github.com/mongoose-os-libs/aws/files/2227196/dht-aws-lowpower-sensor-c.zip)

With Best Regards

**Comment by athind86** [comment]
surround your nan with double quotes, like this "nan" and your final json should be:

`{"state": {"reported": { "dht": { "t": "nan", "h": "nan" } }}, "clientToken": "b0aedebc"}`

**Comment by ghost** [comment]
@athind86 I thank you for your reply. However I am not looking for a way to fix the json. My json is intentionally wrong because I would like to test if the SHADOW library catches the `UPDATE_REJECTED` event raised by AWS. Unfortunately the library doesn't catch it.

**Comment by ghost** [comment]
Issue fixed by #12.
